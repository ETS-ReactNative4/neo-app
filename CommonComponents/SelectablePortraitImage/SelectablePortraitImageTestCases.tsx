import React, { useState } from "react";
import { ITestCase } from "../../TypeInterfaces/TestCases/ITestCases";
import SelectablePortraitImage from "./SelectablePortraitImage";

const SelectablePortraitImageWrapper = () => {
  const [isSelectImage, setSelectImage] = useState<boolean>(false);

  const clickSelectedImage = () => {
    setSelectImage(!isSelectImage);
  };

  return (
    <SelectablePortraitImage
      onPress={() => clickSelectedImage()}
      isSelected={isSelectImage}
    />
  );
};

const SelectablePortraitImageTestCases: ITestCase[] = [
  {
    title: "Selectable Portrait Image",
    Component: <SelectablePortraitImageWrapper />
  }
];

export default SelectablePortraitImageTestCases;
