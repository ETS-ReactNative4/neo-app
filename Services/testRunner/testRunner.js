import renderer from "react-test-renderer";

const testRunner = testCase => {
  test(testCase.title, () => {
    const tree = renderer.create(testCase.Component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export default testRunner;
