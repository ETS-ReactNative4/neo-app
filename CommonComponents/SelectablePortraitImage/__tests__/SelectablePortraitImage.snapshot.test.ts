import SelectablePortraitImageTestCases from "../SelectablePortraitImageTestCases";
import testRunner from "../../../Services/testRunner/testRunner";

SelectablePortraitImageTestCases.map(testRunner);
