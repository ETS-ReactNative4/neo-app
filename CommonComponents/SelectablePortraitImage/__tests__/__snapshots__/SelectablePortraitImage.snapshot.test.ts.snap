// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectable Portrait Image 1`] = `
<View
  style={
    Object {
      "width": 146,
    }
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View>
        <Image
          onError={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "https://d3lf10b5gahyby.cloudfront.net/city/paris.jpg",
            }
          }
          style={
            Array [
              Object {
                "borderRadius": 4,
                "height": 200,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </RNGestureHandlerButton>
</View>
`;
